lib/protoflow-blocks/src/blocks/sys/read_socket.rs

```diff
@@ -0,0 +1,240 @@
+extern crate std;
+use crate::prelude::{vec, String};
+use crate::{IoBlocks, StdioConfig, StdioError, StdioSystem, System};
+use protoflow_core::{
+    types::Any, Block, BlockError, BlockResult, BlockRuntime, Message, OutputPort, Port,
+    PortResult, SystemBuilding,
+};
+use protoflow_derive::Block;
+use serde::{Deserialize, Serialize};
+use simple_mermaid::mermaid;
+use std::{
+    format,
+    io::Read,
+    net::{TcpListener, TcpStream},
+    string::ToString,
+    sync::{Arc, Mutex, PoisonError},
+};
+use tracing::{error, info};
+/// A block that reads a proto object from a TCP port.
+///
+/// # Block Diagram
+#[doc = mermaid!("../../../doc/sys/read_socket.mmd")]
+///
+/// # Sequence Diagram
+#[doc = mermaid!("../../../doc/sys/read_socket.seq.mmd" framed)]
+///
+/// # Examples
+///
+/// ## Using the block in a system
+///
+/// ```rust
+/// # use protoflow_blocks::*;
+/// # fn main() {
+/// System::build(|s| {
+///     // TODO
+/// });
+/// # }
+/// ```
+///
+/// ## Running the block via the CLI
+///
+/// ```console
+/// $ protoflow execute ReadSocket host="127.0.0.1" port="7077" buffer_size="1024"
+/// ```
```

Comment:
Going forward, ideally we would take a connection URL (`tcp://127.0.0.1:7077`) instead of a separate host and port, so that we can support UDP, QUIC, and other protocols as well. But not critical in the current iteration

Reply:
removed host and port, we have only the 'connection' now